pages/checkout/dropin.vue

```diff
@@ -0,0 +1,151 @@
+
+// Reactive references
+const sessionId = ref('');
+const redirectResult = ref('');
+
+// Function to invoke server-side API
+async function sendPostRequest(url, data) {
+  const res = await fetch(url, {
+    method: "POST",
+    body: data ? JSON.stringify(data) : "",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  return await res.json();
+}
+
+// Function to handle the server response
+async function handleServerResponse(res, component) {
+  if (res.action) {
+    component.handleAction(res.action);
+  } else {
+    switch (res.resultCode) {
+      case "Authorised":
+        window.location.href = "/result/success";
+        break;
+      case "Pending":
+      case "Received":
+        window.location.href = "/result/pending";
+        break;
+      case "Refused":
+        window.location.href = "/result/failed";
+        break;
+      default:
+        window.location.href = "/result/error";
+        break;
+    }
+  }
+}
+
+// Function to create AdyenCheckout instance
+async function createAdyenCheckout(session, clientKey) {
+  return AdyenCheckout(
+    {
+      clientKey,
+      session: session,
+      environment: "test",
+      amount: {
+        value: 10000,
+        currency: 'EUR'
+      },
+      locale: "en-US",
+      countryCode: 'NL',
+      showPayButton: true,
+      // override Security Code label
+      translations: {
+        'en-US': {
+            'creditCard.securityCode.label': 'CVV/CVC'
+        }
+      },
+      onPaymentCompleted: (result, component) => {
+        console.info("onPaymentCompleted");
+        console.info(result, component);
+        handleServerResponse(result, component);
+      },
+      onPaymentFailed: (result, component) => {
+        console.info("onPaymentFailed");
+        console.info(result, component);
+        handleServerResponse(result, component);
+      },
+      onError: (error, component) => {
+        console.error("onError");
+        console.error(error.name, error.message, error.stack, component);
+        handleServerResponse(error, component);
+      },
+    }
+  );
+}
+
+// Function to start checkout
+async function startCheckout() {
+  try {
+    const { response, clientKey } = await sendPostRequest(`/api/sessions`);
+
+    const checkout = await createAdyenCheckout(response, clientKey);
+    const dropin = new Dropin(checkout, {
+      paymentMethodsConfiguration: {
+        card: {
+          showBrandIcon: true, // when false not showing the brand logo 
+          hasHolderName: true, // show holder name
+          holderNameRequired: true, // make holder name mandatory
+          // configure placeholders
+          placeholders: { 
+            cardNumber: '1234 5678 9012 3456',
+            expiryDate: 'MM/YY', 
+            securityCodeThreeDigits: '123', 
+            securityCodeFourDigits: '1234',
+            holderName: 'J. Smith'
+          }
+        }
+      },
+      instantPaymentTypes: ['applepay', 'googlepay'],
```

Comment:
Q: Should we include these here?

Reply:
Yes, it is to show express payment methods at the top of the drop-in